src/ngraph/frontend/onnx_import/utils/convpool.hpp

```diff
@@ -92,25 +93,48 @@ namespace ngraph
              * @return nGraph node equivalent of the ONNX operation
              */
             template <class T>
-            inline NodeVector make_ng_pool(const Node& node)
+            inline NodeVector make_ng_pool(const Node& node, bool is_global = false)
             {
                 // Fetch input node for the pooling operation
                 auto data = node.get_ng_inputs().at(0);
 
                 // Parse ONNX op attributes
-                Shape kernel_shape = convpool::get_kernel_shape(node);
+                Shape kernel_shape;
+                if (is_global)
+                {
+                    kernel_shape = node.get_ng_inputs()[0]->get_shape();
+                    kernel_shape.erase(std::begin(kernel_shape), std::next(std::begin(kernel_shape), 2));
+                }
+                else
+                {
+                    kernel_shape = convpool::get_kernel_shape(node);
+                }
                 auto strides = convpool::get_strides(node);
                 auto dilations = convpool::get_dilations(node);
                 auto paddings = convpool::get_pads(node);
 
+                bool count_include_pad = node.get_attribute_value<int64_t>("count_include_pad", 0);
+
                 // Convert padding from CoordinateDiff to Shape objects
                 const CoordinateDiff& padding_above{paddings.first};
                 const CoordinateDiff& padding_below{paddings.second};
                 Shape padding_below_shape{std::begin(padding_below), std::end(padding_below)};
                 Shape padding_above_shape{std::begin(padding_above), std::end(padding_above)};
 
-                return {std::make_shared<T>(
-                    data, kernel_shape, strides, padding_below_shape, padding_above_shape)};
+                if (count_include_pad)
+                {
+                    return {std::make_shared<ngraph::op::AvgPool>(data,
```

Comment:
Can we use `T` instead of `ngraph::op::AvgPool` here?

Reply:
`ngraph::op::MaxPool` dont't have count_include_pad parameter in constructor.